File: bindings/python/examples/pagination.py
# Copyright (c) 2025 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from lib.iota_sdk import *

import asyncio


async def main():
    client = GraphQlClient.new_testnet()
    address = Address.from_hex(
        "0xda1820edf693ee32b5729907b9b2ec8e64980ee8c008c17e89cfb4e5ecd72151")

    all_objects = []
    next_cursor = None
    while True:
        print(f"Fetching page with cursor: {next_cursor}")
        page = await client.objects(
            ObjectFilter(owner=address),
            # Limit to 1 to demonstrate pagination
            PaginationFilter(direction=Direction.FORWARD,
                             cursor=next_cursor,
                             limit=1),
        )
        all_objects.extend(page.data)
        if page.page_info.has_next_page:
            next_cursor = page.page_info.end_cursor
        else:
            break
    print(f"{len(all_objects)} objects fetched:")
    for obj_id in all_objects:
        print(obj_id.id().to_hex())


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/typescript/examples/pagination.ts
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { GraphQlClient, Address, ObjectFilter, PaginationFilter, Direction } from "../lib";

async function main() {
  const client = GraphQlClient.newTestnet();
  const address = Address.fromHex(
    "0xda1820edf693ee32b5729907b9b2ec8e64980ee8c008c17e89cfb4e5ecd72151",
  );

  const allObjects: any[] = [];
  let nextCursor: string | undefined = undefined;
  while (true) {
    console.log(`Fetching page with cursor: ${nextCursor}`);
    const page = await client.objects(
      ObjectFilter.create({ owner: address }),
      // Limit to 1 to demonstrate pagination
      PaginationFilter.create({ direction: "forward", cursor: nextCursor, limit: 1 }),
    );
    allObjects.push(...page.data);
    if (page.pageInfo.hasNextPage) {
      nextCursor = page.pageInfo.endCursor;
    } else {
      break;
    }
  }
  console.log(`${allObjects.length} objects fetched:`);
  for (const obj of allObjects) {
    console.log(obj.objectId().toHex());
  }
}

main();
